overlay/usr/lib/inithooks/bin/domain-controller.py

```diff
@@ -157,42 +182,64 @@ def rm_glob(path):
         rm_f(file_path)
 
 
-def run_command(command):
-    proc = subprocess.Popen(command, encoding='utf-8',
-                            stdout=PIPE, stderr=STDOUT)
-    output = []
-    while True:
-        out = proc.stdout.read(1)
-        if out == '' and proc.poll() is not None:
-            break
-        if out != '':
-            output.append(out)
-            sys.stdout.write(out)
-            sys.stdout.flush()
-    return proc.returncode, "".join(output)
-
-
-def update_resolvconf(domain):
-    resolvconf_head = '/etc/resolvconf/resolv.conf.d/head'
-    with open(resolvconf_head, 'r') as fob:
+def run_command(command, stdin=False):
+    if not command:
+        return 0, None
+    if stdin:
+        proc = subprocess.Popen(command, text=True, stdin=PIPE,
+                                stdout=PIPE, stderr=STDOUT)
+    else:
+        proc = subprocess.Popen(command, text=True,
+                                stdout=PIPE, stderr=STDOUT)
```

Comment:
Could be simplified with ternary:
`proc = subprocess.Popen(command, text=True, stdin=PIPE if stdin else None, stdout=PIPE, stderr=STDOUT)`

Reply:
Thanks for this. FWIW, I had already improved that code a little, but I still implemented our suggestion.